### PR TITLE
Favorite Food Items with Toggle

### PR DESCRIPTION
**Pull request by leo-he-1**

### CODE REVIEW

File: mobile/src/screens/favoriteFoodItems/FavoriteFoodItems.styles.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    title: {
+        fontSize: size.lg,
+        fontFamily: font.semiBold,
+        color: color.black,
+        paddingVertical: size.xs,
+    },
+    button: {
+        padding: size.rg,
+        borderRadius: size.xxl,
+        width: 160,
```
Comment (lasharela) [inline mobile/src/screens/favoriteFoodItems/FavoriteFoodItems.styles.ts:25]: Not fixed? fixed?

**Comment by leo-he-1** [comment]
image

**Comment by leo-he-1** [comment]
image

File: mobile/src/screens/favoriteFoodItems/FavoriteFoodItems.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  },
+];
+
+
+export const FavoriteFoodItems: FC<TFavoriteFoodItemsNavigation> = ({navigation}) => {
+  const handleOnPress = () => {
+    console.log('navigate to screen');
+  };
+
+  const [option, setOption] = useState<boolean>(true)
```
Comment (lasharela) [inline mobile/src/screens/favoriteFoodItems/FavoriteFoodItems.tsx:79]: set more descriptive name pls

File: mobile/src/screens/favoriteFoodItems/FavoriteFoodItems.type.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { StackScreenProps } from '@react-navigation/stack';
+import { Dispatch, SetStateAction } from 'react';
+import { TAppNavigationRoutes, EAppNavigationRoutes, TAppNavigation } from '../../navigation';
+
+export type TFavoriteFoodItemsScreenProps = StackScreenProps<TAppNavigationRoutes, EAppNavigationRoutes.FAVORITEFOODITEMS>;
+
+export type TFavoriteFoodItemsScreenViewProps = {
+    onPress: () => void;
+    favoriteFoodItems:Array<Object>
```
Comment (lasharela) [inline mobile/src/screens/favoriteFoodItems/FavoriteFoodItems.type.ts:9]: [TMyFoodItems]
+ TMyFoodItems[]
+ String[]

File: mobile/src/screens/favoriteFoodItems/FavoriteFoodItems.view.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  favoriteFoodItems,
+  favoriteResturants,
+  onPress,
+  isLoading = false,
+  showFavoriteFoods= true,
+  setShowFavoriteFoods,
+}) => {
+
+    return (
+        <Container isLoading={isLoading}>
```
Comment (lasharela) [inline mobile/src/screens/favoriteFoodItems/FavoriteFoodItems.view.tsx:21]: <Container isLoading={isLoading} header={...componentHere}>

**Review by lasharela — CHANGES_REQUESTED** [review]

**Review by Josetf2396 — APPROVED** [review]

**Review by Dulaya — APPROVED** [review]

**Comment by Dulaya** [comment]
- In FavoriteFoodItems.view, you have an un used onPress. Add onPress={onPress} to TouchableOpacity.

Optional: 

 - If FavoriteFoodItems.tsx, you can pass this code as prop to FavoriteFoodItemScreenView:
 - 
            leftPress={() => navigation.goBack()}

      - Make sure you add to the type file with type () => void
   - In Header, you and do leftPress={leftPress}
      - This will cause an error for now because there's no route to back to anything yet.

 - In FavoriteFoodItems.view, remove unnecessary codes (View, return). This code work same as your code: 

            {(showFavoriteFoods ? favoriteFoodItems : favoriteResturants).map((food, index) => 
               <TouchableOpacity key={String(index)}>
                  <FoodCard foodItem={food} />
               </TouchableOpacity>
            )}

 - In Header, you can pass picture like this instead of using rightIconName:
      rightIconLocation="https://i.imgur.com/EQn7qGf.png"

**Review by Nap1400 — APPROVED** [review]

**Pull request merged**